testSuite/scripts/test_service_to_service_copy.py

```diff
@@ -1147,12 +1147,17 @@ def util_test_copy_file_from_x_bucket_to_x_bucket_propertyandmetadata(
         validate_dir_name = "validate_copy_file_from_%s_bucket_to_%s_bucket_propertyandmetadata_%s" % (srcType, dstType, preserveProperties)
         local_validate_dest_dir = util.create_test_dir(validate_dir_name)
         local_validate_dest = local_validate_dest_dir + fileName
+        # Because the MD5 is checked early, we need to clear the check-md5 flag.
         if srcType == "S3":
             result = util.Command("copy").add_arguments(dstFileURL).add_arguments(local_validate_dest). \
-                add_flags("log-level", "info").execute_azcopy_copy_command()
+                add_flags("log-level", "info")  # Temporarily set result to Command for the sake of modifying the md5 check
+            result.flags["check-md5"] = ""
+            result = result.execute_azcopy_copy_command()  # Wrangle result to a bool for checking
         else:
             result = util.Command("copy").add_arguments(srcFileURL).add_arguments(local_validate_dest). \
-                add_flags("log-level", "info").execute_azcopy_copy_command()
+                add_flags("log-level", "info")  # Temporarily set result to Command for the sake of modifying the md5 check
+            result.flags["check-md5"] = ""
+            result = result.execute_azcopy_copy_command()  # Wrangle result to a bool for checking
```

Comment:
I haven't quite understood what's going on here.  Specifically, anything that fails now should have failed before - right?  (All this PR does is move the failure forward in time).  So:
(a) why is this change necessary? (Since anything that fails now must have also failed before).
(b) if this change is necessary, why is it only necessary in this one test (and not many tests)?

Reply:
a) When we choose not to persist the properties and metadata about the file, we lose the MD5. The MD5 is set to "FailIfDifferentOrMissing", so the file transfer never takes place, and these tests fail.

b) This is the only S2S test utility that opts to not copy properties and metadata occasionally. Otherwise, we're always putting and checking MD5s.